### PR TITLE
Add invite by email screen in create space flow

### LINKED ISSUE #3678: Spaces | M3.23 Invite by email in private space
https://www.figma.com/file/m7L63aGPW7iHnIYStfdxCe/Spaces?node-id=2601%3A2385

image

Depends on https://github.com/vector-im/element-android/issues/548 (part of it should be fixed for a better experience)

### LINKED ISSUE #3945: Remove the "Teammate spaces aren't quite ready" bottom sheet
### Steps to reproduce

1. Left panel -> New Space
2. "Private"
3. Name the space, then click "NEXT"
4. "Me and teammates"

### What did you expect?

Get taken to the "create first rooms" view

### What happened?

See a bottom sheet warning me about the mode being experimental.

### Application version and app store

https://buildkite.com/organizations/matrix-dot-org/pipelines/element-android/builds/3224/jobs/a1fe572d-8a7b-4406-8a0f-37f4c9ff4787/artifacts/30990f61-8f7d-4dc6-af1e-340765110cc6

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #3678 (add invite by mail screen for private space)
Fixes #3945 Remove the "Teammate spaces aren't quite ready" bottom sheet

When creating a new private team space, a screen is added where you can add up tp 3 emails. There is no integration with contact book for now. 
- If no emails entered you can skip this step
- If one the email is malformed an inline error is displayed and you have to fix in order to continue
- If no Identity Server is defined, user will see a message and an action to open discovery settings 

image

image

image

### CODE REVIEW

**Comment by BillCarsonFr** [comment]
Full flow when you don't have IDS
image

File: vector/src/main/java/im/vector/app/features/spaces/create/CreateSpaceAdd3pidInvitesFragment.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    override fun onBackPressed(toolbarButton: Boolean): Boolean {
+        sharedViewModel.handle(CreateSpaceAction.OnBackPressed)
+        return true
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        views.recyclerView.configureWith(epoxyController)
+        epoxyController.listener = this
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/spaces/create/CreateSpaceAdd3pidInvitesFragment.kt:50]: Add cleanup in `onDestroyView()`

File: vector/src/main/java/im/vector/app/features/spaces/create/CreateSpaceViewModel.kt [hunk 2]
```diff
@@ -49,19 +51,48 @@ class CreateSpaceViewModel @AssistedInject constructor(
...
     interface Factory {
         fun create(initialState: CreateSpaceState): CreateSpaceViewModel
     }
 
+    private fun startListenToIdentityManager() {
+        identityService.addListener(identityServerManagerListener)
+    }
+
+    private fun stopListenToIdentityManager() {
+        identityService.addListener(identityServerManagerListener)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/spaces/create/CreateSpaceViewModel.kt:88]: removeListener :)

**Review by bmarty — CHANGES_REQUESTED** [review]
(static review) 2 small remarks

File: vector/src/main/java/im/vector/app/features/spaces/create/CreateSpaceAdd3pidInvitesFragment.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/spaces/create/CreateSpaceAdd3pidInvitesFragment.kt:50]: Done

File: vector/src/main/java/im/vector/app/features/spaces/create/CreateSpaceViewModel.kt (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/spaces/create/CreateSpaceViewModel.kt:88]: 🙃

**Review by bmarty — APPROVED** [review]
Thanks for the update.

**Pull request merged**